rdrf/report/schema.py

```diff
@@ -84,6 +84,26 @@ class Meta:
         fields = ('name', 'code')
 
 
+class ParentGuardianType(DjangoObjectType):
+    email = graphene.String()
+    self_patient_id = graphene.Int()
+    gender = graphene.String()
+
+    class Meta:
+        model = ParentGuardian
+        fields = ('first_name', 'last_name', 'date_of_birth', 'place_of_birth', 'date_of_migration', 'address',
+                  'suburb', 'state', 'postcode', 'country', 'phone')
+
+    def resolve_email(parent_guardian, info):
+        return parent_guardian.user.email
+
+    def resolve_self_patient_id(parent_guardian, info):
+        return parent_guardian.self_patient.id
```

Comment:
`self_patient` may be `None`

Reply:
For some reason I thought I tested this and it worked (which doesn't make sense), but you're right, it throws an error.

Reply:
Interestingly, it does throw an error, but it still returns the data, so that explains why the CSV and JSON extract still appeared to work:

```
{
  "errors": [
    {
      "message": "'NoneType' object has no attribute 'id'",
      "locations": [
        {
          "line": 23,
          "column": 7
        }
      ],
      "path": [
        "patients",
        0,
        "parentguardianSet",
        0,
        "selfPatientId"
      ]
    }
  ],
  "data": {
    "dataSummary": {
      "maxParentGuardianCount": 1
    },
    "patients": [
      {
        "familyName": "BROWN",
        "givenNames": "Karen0",
        "ethnicOrigin": null,
        "parentguardianSet": [
          {
            "firstName": "Gayle",
            "lastName": "White",
            "dateOfBirth": "2022-05-25",
            "placeOfBirth": "Brisbane",
            "dateOfMigration": "2022-05-25",
            "gender": "Female",
            "address": "123 Alpha Street",
            "suburb": "Brisbane",
            "state": "QLD",
            "postcode": "4000",
            "country": "Australia",
            "phone": "0412123456",
            "email": "user@test.com",
            "selfPatientId": null
          }
        ]
      },
```